frontend/src/metabase/query_builder/components/view/SnippetSidebarButton.jsx

```diff
@@ -0,0 +1,26 @@
+import React from "react";
+
+import { t } from "ttag";
+import cx from "classnames";
+
+import Icon from "metabase/components/Icon";
+import Tooltip from "metabase/components/Tooltip";
+
+const SnippetSidebarButton = ({
+  toggleSnippetSidebar,
+  isShowingSnippetSidebar,
+  className,
+  size,
+}) => (
+  <Tooltip tooltip={t`SQL Snippets`}>
+    <a
+      className={cx(className, "transition-color text-brand-hover", {
+        "text-brand": isShowingSnippetSidebar,
+      })}
+    >
+      <Icon name="snippet" size={size} onClick={toggleSnippetSidebar} />
```

Comment:
(Aside, I'm not a big fan of this icon representing snippets, but that's a design issue rather than a code issue)